File: OOPJava/src/main/game/rosliny/Guarana.java
package main.game.rosliny;

import main.game.*;
public class Guarana extends Roslina{
    public Guarana(int x, int y, Swiat swiat){ super(0,x,y,swiat); }
    public Guarana(int sila, int inicjatywa, int x, int y, int poprzednieX, int poprzednieY, int wiek, Swiat swiat) {
        super(sila, inicjatywa, x, y, poprzednieX, poprzednieY, wiek, swiat);
    };

    @Override
    public String rysowanie(){
        return "guarana.png";
    }
    @Override
    public void kolizja(Organizm atakujacy){
        atakujacy.setSila(atakujacy.getSila() + 3);//zwieksza sile organizmu, ktory ja je o 3
        swiat.zabijOrganizm(this);
        swiat.dodajLog(atakujacy.getImie() + " zjadl " + this.getImie());
        return;
    }
    @Override
    public void rozmnozSie(int x, int y, Swiat swiat){
        Guarana nowyOrganizm = new Guarana(x, y, swiat);
        swiat.dodajOrganizm(nowyOrganizm);
    }

    @Override
    public String getImie() { return "Guarana"; };
}

File: PROJECT_C++/PROJECT_C++/Guarana.cpp
#include "Guarana.h"
#include <iostream>
#include "Swiat.h"
#include <string>
#include <vector>

Guarana::Guarana(int x, int y, Swiat& swiat) : Roslina(0, x, y, swiat) {};

void Guarana::rysowanie() {
	std::cout << "+";
}

void Guarana::kolizja(Organizm& atakujacy) {//domyslnie roslija przegrywa kazdy pojedynek
	atakujacy.setSila(atakujacy.getSila() + 3);//zwieksza sile organizmu, ktory ja je o 3
	swiat.zabijOrganizm(*this);
	swiat.dodajLog(atakujacy.getImie() + " zjadl " + this->getImie());
	return;
}

void Guarana::rozmnozSie(int x, int y, Swiat& swiat) {
	Guarana* nowyOrganizm = new Guarana(x, y, swiat);
	swiat.dodajOrganizm(*nowyOrganizm);
}

Guarana::~Guarana() {};